### PR TITLE
Remove System.Collections.Immutable dependency where not needed

### LINKED ISSUE #59: System.Collections.Immutable should not be an inbox dependency for components
### Describe the bug

Found in https://github.com/CommunityToolkit/Labs-Windows/pull/438

> @Arlodotexe I think your infra is injecting extra dependencies into packages, look at this:
> 
> 
> 
> I'm not WinUI nor the `Immutable` package anywhere, and they shouldn't be there. Could you take a look? 🙂

While this package is required, it does not need to be transient to projects that reference this package.

### Steps to reproduce

```text
n/a
```

### Expected behavior

n/a

### Screenshots

_No response_

### Code Platform

- [ ] UWP
- [ ] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

_No response_

### Additional context

_No response_

### Help us help you

Yes, I'd like to be assigned to work on this item.

### PR DESCRIPTION
**Pull request by Arlodotexe**
Closes #59

### CODE REVIEW

**Comment by Sergio0694** [comment]
Why "non-transient" in the title? 🤔

**Comment by Arlodotexe** [comment]
> Why "non-transient" in the title? 🤔

Safest option. I don't think we need this package at on `netstandard2.0` tho, so let's pivot a bit.

**Comment by Sergio0694** [comment]
> "I don't think we need this package at all on netstandard2.0 tho"

We shouldn't be importing this package by default _on any target_.

**Comment by Arlodotexe** [comment]
Looks like this was originally added here to resolve MSB4011 warnings - https://github.com/CommunityToolkit/Labs-Windows/pull/117/files#diff-111cb6ee41530a23ad2012cac97291a76f7a52498bb200d707051bf05d376c14R6

A [short time after](https://github.com/CommunityToolkit/Labs-Windows/issues/99#issuecomment-1240996231) we found out this was [a bug in Roslyn](https://github.com/dotnet/roslyn/issues/63780), which appears to be resolved now. Let's get this cleaned up. Thanks @Sergio0694!

File: ToolkitComponent.SampleProject.props [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
   </Target>
-  
+
   <!-- Set up the MultiTarget system -->
   <Import Project="$(ToolingDirectory)\MultiTarget\Library.props" />
 
   <!-- Import this component's source project -->
   <ItemGroup>
     <ProjectReference Include="$(MSBuildProjectDirectory)\..\src\*.csproj" />
+    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" PrivateAssets="All" />
```
Comment (Sergio0694) [inline ToolkitComponent.SampleProject.props:12]: Don't think this needs private assets, if it's just the sample project?

**Review by Sergio0694 — APPROVED** [review]
One nit, otherwise LGTM!

**Pull request merged**